persistent/tests/test_timestamp.py

```diff
@@ -271,26 +276,37 @@ def test_py_hash_32_64_bit(self):
             py = self._makePy(*self.now_ts_args)
             self.assertEqual(hash(py), bit_32_hash)
 
+
             persistent.timestamp.c_long = ctypes.c_int64
             # call __hash__ directly to avoid interpreter truncation
             # in hash() on 32-bit platforms
-            self.assertEqual(py.__hash__(), bit_64_hash)
+            if not _is_jython:
+                self.assertEqual(py.__hash__(), bit_64_hash)
+            else:
+                # Jython 2.7's ctypes module doesn't properly
+                # implement the 'value' attribute by truncating.
+                # (It does for native calls, but not visibly to Python).
+                # Therefore we get back the full python long. The actual
+                # hash() calls are correct, though, because the JVM uses
+                # 32-bit ints for its hashCode methods.
```

Comment:
Oh. I don't think that `__hash__` is required to truncate. Only `hash()` is. From `object.__hash__`:

> Changed in version 2.5: `__hash__()` may now also return a long integer object; the 32-bit integer is then derived from the hash of that object.

`Timestamp.__hash__` uses ctypes to truncate to a 64-bit value, but ctypes fails to do so under jython. They know it's broken and are rewriting it.